network/file_output.cpp

```diff
@@ -55,11 +61,49 @@ FileOutput::FileOutput(VideoOptions const *options) : Output(options) {
     fileNameGenerator << latestDir_;
     fileNameGenerator << "latest.txt";
     latestFileName_ = fileNameGenerator.str();
+
+    collectExistingFilenames();
 }
 
 FileOutput::~FileOutput() {
 }
 
+void FileOutput::collectExistingFilenames() {
+    fileQueueMutex_.lock();
```

Comment:
I see there's a C++ construct called a std::lock_guard that manages the mutex locking using RAII. I think I'll change the code to use that because I'm worried about exceptions creating a deadlock.